src/tensorboard_logger.cc

```diff
@@ -98,18 +98,34 @@ int TensorBoardLogger::add_histogram(const std::string &tag, int step,
 }
 
 int TensorBoardLogger::add_histogram(const string &tag, int step,
-                                     const vector<float> &values) {
+                                     const vector<float> &values_) {
+    std::vector<double> values(values_.begin(), values_.end());
```

Comment:
Creating a new vector will incur some penalty here, especially when the vector size is large, which is common in large DNN models. Actually the reason why I added a `add_histogram(const std::string &tag, int step, const float *value, size_t num) ` signature is also to avoid creating vectors, and it really helped in our scenario. Maybe a better choice is to use templates?